packages/synapse-react-client/src/synapse-queries/entity/useEntity.ts

```diff
@@ -229,166 +251,159 @@ export function removeStandardEntityFields(
 export function useGetJson(
   entityId: string,
   includeDerivedAnnotations: boolean,
-  options?: UseQueryOptions<EntityJson, SynapseClientError>,
+  options?: Partial<UseQueryOptions<EntityJson, SynapseClientError>>,
 ) {
   const { accessToken, keyFactory } = useSynapseContext()
-  const query = useQuery<EntityJson, SynapseClientError>(
-    keyFactory.getEntityJsonQueryKey(entityId, includeDerivedAnnotations),
-    () =>
+  return useQuery({
+    ...options,
+    queryKey: keyFactory.getEntityJsonQueryKey(
+      entityId,
+      includeDerivedAnnotations,
+    ),
+    queryFn: () =>
       SynapseClient.getEntityJson(
         entityId,
         includeDerivedAnnotations,
         accessToken,
       ),
-    options,
-  )
 
-  // Separate the standard fields and annotations.
-  const entityMetadata = useMemo(
-    () =>
-      query?.data == undefined
-        ? undefined
-        : getStandardEntityFields(query.data),
-    [query.data],
-  )
+    select: data => {
+      // Separate the standard fields and annotations.
+      const entityMetadata = getStandardEntityFields(data)
 
-  const annotations = useMemo(
-    () =>
-      query?.data == undefined
-        ? undefined
-        : removeStandardEntityFields(query.data),
-    [query.data],
-  )
+      const annotations = removeStandardEntityFields(data)
 
-  return {
-    ...query,
-    entityMetadata,
-    annotations,
-  }
+      return {
+        entity: data,
+        entityMetadata,
+        annotations,
+      }
+    },
+  })
```

Comment:
Transform the data using `select` instead of appending fields to the query result